use-cases/demos/object_detection/pytorch-yolo/entrypoint.sh

```diff
@@ -9,7 +9,23 @@
 cid=$(date +%Y%m%d%H%M%S%N)
 echo "cid: $cid"
 
-PROFILE_NAME="demo_pytorch_object_detection"
+echo "USE_ULTRALYTICS: $USE_ULTRALYTICS"
+echo "INTEL_OPTIMIZED: $INTEL_OPTIMIZED"
 
-python3 object_detection/pytorch-yolo/yolov5_object_detection.py \
+if [ $USE_ULTRALYTICS -eq 1 ]; 
+then
+ if [ $INTEL_OPTIMIZED -eq 1 ]; 
+ then 
+ PROFILE_NAME="demo_intel_ultralytics_pytorch_object_detection"
+ python3 object_detection/pytorch-yolo/yolov5_ultralytics_object_detection.py \
+-m "$MQTT_HOSTNAME" -p "$MQTT_PORT" -t "$MQTT_TOPIC" -i "$INPUT_SRC" -in | tee >/tmp/results/r"$cid"_"$PROFILE_NAME".jsonl >(stdbuf -oL sed -n -e 's/^.*FPS: //p' | stdbuf -oL cut -d , -f 1 > /tmp/results/pipeline"$cid"_"$PROFILE_NAME".log)
+ else
+ PROFILE_NAME="demo_ultralytics_pytorch_object_detection"
+ python3 object_detection/pytorch-yolo/yolov5_ultralytics_object_detection.py \
 -m "$MQTT_HOSTNAME" -p "$MQTT_PORT" -t "$MQTT_TOPIC" -i "$INPUT_SRC" | tee >/tmp/results/r"$cid"_"$PROFILE_NAME".jsonl >(stdbuf -oL sed -n -e 's/^.*FPS: //p' | stdbuf -oL cut -d , -f 1 > /tmp/results/pipeline"$cid"_"$PROFILE_NAME".log)
```

Comment:
there is duplicated code, could be simplify:
```suggestion
 if [ $INTEL_OPTIMIZED -eq 1 ]; 
 then 
 PROFILE_NAME="demo_intel_ultralytics_pytorch_object_detection"
 else
 PROFILE_NAME="demo_ultralytics_pytorch_object_detection"
 fi
 
 python3 object_detection/pytorch-yolo/yolov5_ultralytics_object_detection.py \
-m "$MQTT_HOSTNAME" -p "$MQTT_PORT" -t "$MQTT_TOPIC" -i "$INPUT_SRC" | tee >/tmp/results/r"$cid"_"$PROFILE_NAME".jsonl >(stdbuf -oL sed -n -e 's/^.*FPS: //p' | stdbuf -oL cut -d , -f 1 > /tmp/results/pipeline"$cid"_"$PROFILE_NAME".log)
```

Reply:
no..python file is called with different option here -in for INTEL_OPTIMIZED=1 and no option if it INTEL_OPTIMIZED=0

Reply:
why can't it be parametrized to -in 1 or 0? eg. -in ${INTEL_OPTOMIZED} so the code will be the same?

Reply:
also what happens to the case USE_ULTRALYTICS = 0 and INTEL_OPTIMIZED =1? the code is not very obvious to indicate that case

Reply:
If code is USE_ULTRALYTICS = 0, it will enter into outer else. Doesn't matter Intel_optimzed=1 or 0. I can add these details in documentation + comments in code.

Reply:
> why can't it be parametrized to -in 1 or 0? eg. -in ${INTEL_OPTOMIZED} so the code will be the same?

argparse store true works in this way

Reply:
you mean if not use_ultralytics, it is always intel optimized???

Reply:
> > why can't it be parametrized to -in 1 or 0? eg. -in ${INTEL_OPTOMIZED} so the code will be the same?
> 
> argparse store true works in this way

doesn't have to be in this way

Reply:
change it not using store true or just take parameter and you can do the parsing logic in the code

Reply:
> you mean if not use_ultralytics, it is always intel optimized???

not use_ultralytics flow doesn't support intel optimized at all. So doesn't affect here.

Reply:
> change it not using store true or just take parameter and you can do the parsing logic in the code

its an implementation choice, i want to use boolean instead of using string comparison in code since then we need to take care of uppercase/lowercase/spelling mistakes etc. Unfortunately bash doesn't support bool datatype so end up using int comparison. Argparse implements bool using store true. I didn't find any other way to input boolean input parameters using argparse other than using string comparison which i want to avoid.

Reply:
I agree that using -eq vs "==" makes sense in this context. 

You could add a DETECTION_SCRIPT variable that will change based on the if logic above. Updating the python3 call to use the $DETECTION_SCRIPT the redundant code.

Reply:
> I agree that using -eq vs "==" makes sense in this context.
> 
> You could add a DETECTION_SCRIPT variable that will change based on the if logic above. Updating the python3 call to use the $DETECTION_SCRIPT the redundant code.

that is a good suggestion, let me update.

Reply:
code duplication is fixed based on Brian's suggestion. All flows worked on testing.